File: Another Sorting Problem.cpp
//https://codeforces.com/contest/1575/problem/A
#include <bits/stdc++.h>
using namespace std;

char inverseChar(char c) {
    char cRet=26-(c-64)+1+64;
    return cRet;
}
int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);cout.tie(NULL);
    int n,m;
    char c;
    vector<string> v;
    map<string,int> mp;
    cin>>n>>m;
    for(int i=1;i<=n;i++) {
        deque<char> d;
        for(int j=1;j<=m;j++) {
            cin>>c;
            if(j%2) d.push_back(c);
            else d.push_back(inverseChar(c));
        }
        string s{d.begin(),d.end()};
        mp[s]=i;
        v.push_back(s);
    }
    sort(v.begin(),v.end());
    for(auto x:v) cout<<mp[x]<<" ";
}

File: Another Sorting Problem.java
//https://codeforces.com/contest/1575/problem/A
import java.util.*;
public class Another_Sorting_Problem{
    static char inverseChar(char c) {
        int x=26-(c-64)+1+64;
        char cRet=(char) x;
        return cRet;
    }
    public static void main(String[] args) {
        int n,m;
        String s,t;
        char c;
        Scanner sc=new Scanner(System.in);
        n=sc.nextInt();
        m=sc.nextInt();
        Vector<String> v=new Vector(n);
        Map<String,Integer> mp=new HashMap<>();
        for(int i=1;i<=n;i++) {
            s=sc.next(); 
            t="";
            for(int j=1;j<=m;j++) {
                c=s.charAt(j-1);
                if(j%2>0) t+=c;
                else t+=inverseChar(c);
            }
            mp.put(t,i);
            v.add(t);
        }
        Collections.sort(v);
        for(String x:v) System.out.print(mp.get(x)+" ");
    }
}